activesupport/lib/active_support/core_ext/module/attribute_accessors.rb

```diff
@@ -64,14 +60,16 @@ def self.#{sym}
         end
       EOS
 
-      unless options[:instance_reader] == false || options[:instance_accessor] == false
+      if instance_reader && instance_accessor
         class_eval(<<-EOS, __FILE__, __LINE__ + 1)
           def #{sym}
             @@#{sym}
           end
         EOS
       end
-      class_variable_set("@@#{sym}", yield) if block_given?
+
+      default = yield if block_given? && default.nil?
```

Comment:
We had a test for that even before, see [this](https://github.com/rails/rails/blob/f7388635f246b97a1aa0be54579339c6659fb865/activesupport/test/core_ext/module/attribute_accessor_test.rb#L94-L98).